### PR TITLE
Feature/add private cluster support

### PR DESCRIPTION
**Pull request by askulkarni2**
*Issue #, if available:* #74

*Description of changes:*
Adding support for private clusters. `ec2-cluster-provider` now has an additional property `privateCluster?: boolean` which when set to true, creates a private only cluster, i.e. cluster API is private and the cluster is private with nodegroups provisioned in private subnets. Also added new sample stack to demonstrate usage.

Will submit a separate PR for SSM addon.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/cluster-providers/ec2-cluster-provider.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ export interface EC2ProviderClusterProps extends CommonClusterOptions {
      * Select either SPOT or ON-DEMAND
      */
     nodeGroupCapacityType?: CapacityType;
+
+    /**
+     * Is it a private only EKS Cluster
+     *
+     * @default true
```
Comment (shapirov103) [inline lib/cluster-providers/ec2-cluster-provider.ts:81]: I think the default should stay private and public with an option to override.

**Review by shapirov103 — COMMENTED** [review]
Validation passed? All addons were installed successfully?

**Review by kcoleman731 — COMMENTED** [review]
LGTM. Agree with Mikhail's comment.

**Comment by askulkarni2** [comment]
> Validation passed? All addons were installed successfully?

Yes. Validated.

```sh
kubectl get pods -ANAMESPACE           NAME                                                              READY   STATUS    RESTARTS   AGE
amazon-cloudwatch   cloudwatch-agent-8j8j9                                            1/1     Running   0          161m
amazon-cloudwatch   fluentd-cloudwatch-mltt9                                          1/1     Running   0          161m
argocd              privateclusterblueprintchartargocdaddona13e3f4b-application7r7p   1/1     Running   0          161m
argocd              privateclusterblueprintchartargocdaddona13e3f4b-dex-serverl76p7   1/1     Running   0          161m
argocd              privateclusterblueprintchartargocdaddona13e3f4b-redis-99f9shsz4   1/1     Running   0          161m
argocd              privateclusterblueprintchartargocdaddona13e3f4b-repo-servex694d   1/1     Running   0          161m
argocd              privateclusterblueprintchartargocdaddona13e3f4b-server-6bbk4b5n   1/1     Running   0          161m
kube-system         aws-node-4cbtx                                                    1/1     Running   0          165m
kube-system         calico-node-594fw                                                 1/1     Running   0          161m
kube-system         calico-typha-55d95897b8-vr86g                                     1/1     Running   0          161m
kube-system         calico-typha-horizontal-autoscaler-767b5c958c-h6f7b               1/1     Running   0          161m
kube-system         coredns-6548845887-jch4d                                          1/1     Running   0          172m
kube-system         coredns-6548845887-prgqd                                          1/1     Running   0          172m
kube-system         kube-proxy-8nt6p                                                  1/1     Running   0          165m
kube-system         metrics-server-5d5c49f488-d695p                                   1/1     Running   0          161m
kube-system         privateclusterblueprintchartngninxaddon50fa0804-nginx-ingrlv67b   1/1     Running   0          161m
```

File: lib/cluster-providers/ec2-cluster-provider.ts (hunk 1 above)
Comment (shapirov103) [reply lib/cluster-providers/ec2-cluster-provider.ts:81]: Let's resolve this, otherwise lgtm, nice!

Comment (askulkarni2) [reply lib/cluster-providers/ec2-cluster-provider.ts]: Resolved.

**Review by shapirov103 — APPROVED** [review]
lgtm

**Pull request merged**